internal/portforward/service/service.go

```diff
@@ -47,3 +47,45 @@ func (s *Service) GetPortsForwarded() (ports []uint16) {
 	copy(ports, s.ports)
 	return ports
 }
+
+func (s *Service) SetPortsForwarded(ctx context.Context, ports []uint16) (err error) {
+	for i, port := range s.ports {
+		err := s.portAllower.RemoveAllowedPort(ctx, port)
+		if err != nil {
+			for j := 0; j < i; j++ {
+				_ = s.portAllower.SetAllowedPort(ctx, s.ports[j], s.settings.Interface)
+			}
+			s.logger.Error(err.Error())
+			return err
+		}
+	}
+
+	for i, port := range ports {
+		err := s.portAllower.SetAllowedPort(ctx, port, s.settings.Interface)
+		if err != nil {
+			for j := 0; j < i; j++ {
+				_ = s.portAllower.RemoveAllowedPort(ctx, s.ports[j])
+			}
+			for _, port := range s.ports {
+				_ = s.portAllower.SetAllowedPort(ctx, port, s.settings.Interface)
+			}
+			s.logger.Error(err.Error())
+			return err
```

Comment:
Let's remove the log here and let the caller handle the error

```suggestion
			return fmt.Errorf("setting allowed port: %w", err)
```

Reply:
Fixed with 25fd6ff00ae7996c6876ae45b02e533f29024d18